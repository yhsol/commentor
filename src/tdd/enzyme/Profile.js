import React from "./node_modules/react";

const Profile = ({ username, name }) => {
  return (
    <div>
      <b>{username}</b>&nbsp;
      <span>({name})</span>
    </div>
  );
};

export default Profile;
