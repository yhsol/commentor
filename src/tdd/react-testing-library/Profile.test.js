import React from "react";
import Profile from "./Profile";
import { render } from "@testing-library/react";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile username="username" name="name" />);
    expect(utils.container).toMatchSnapshot();
  });
  it("shows the props correctly", () => {
    const utils = render(<Profile username="username" name="name" />);
    utils.getByText("username");
    utils.getByText("(name)");
    utils.getAllByText(/name/);
  });
});
