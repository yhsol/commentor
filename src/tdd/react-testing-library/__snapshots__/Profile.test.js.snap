// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      username
    </b>
    <span>
      (
      name
      )
    </span>
  </div>
</div>
`;
